### PR TITLE
task_wdt: add shell

### PR DESCRIPTION
**Pull request by maass-hamburg**
add shell for task watchdog, mainly for testing

### CODE REVIEW

File: subsys/task_wdt/task_wdt_shell.c [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		shell_fprintf(sh, SHELL_ERROR, "Failed to init task watchdog: %d\n", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int cmd_add(const struct shell *sh, size_t argc, char **argv)
+{
+	if (argc < 2) {
```
Comment (martinjaeger) [inline subsys/task_wdt/task_wdt_shell.c:37]: Does this not need *exactly* one parameter?

```suggestion
	if (argc != 2) {
```

File: subsys/task_wdt/task_wdt_shell.c [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	}
+
+	shell_fprintf(sh, SHELL_INFO, "Task watchdog channel: %d\n", ret);
+
+	return 0;
+}
+
+static int cmd_feed(const struct shell *sh, size_t argc, char **argv)
+{
+	if (argc < 2) {
```
Comment (martinjaeger) [inline subsys/task_wdt/task_wdt_shell.c:60]: Same here:
```suggestion
	if (argc != 2) {
```

File: subsys/task_wdt/task_wdt_shell.c [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+		shell_fprintf(sh, SHELL_ERROR, "Failed to add task watchdog channel: %d\n", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int cmd_del(const struct shell *sh, size_t argc, char **argv)
+{
+	if (argc < 2) {
```
Comment (martinjaeger) [inline subsys/task_wdt/task_wdt_shell.c:79]:
```suggestion
	if (argc != 2) {
```

File: subsys/task_wdt/task_wdt_shell.c [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+		shell_fprintf(sh, SHELL_ERROR, "Invalid number of arguments\n");
+		return -EINVAL;
+	}
+
+	shell_fprintf(sh, SHELL_INFO, "Delete task watchdog channel %s\n", argv[1]);
+
+	int ret = task_wdt_delete(atoi(argv[1]));
+
+	if (ret < 0) {
+		shell_fprintf(sh, SHELL_ERROR, "Failed to add task watchdog channel: %d\n", ret);
```
Comment (martinjaeger) [inline subsys/task_wdt/task_wdt_shell.c:89]:
```suggestion
		shell_fprintf(sh, SHELL_ERROR, "Failed to delete task watchdog channel: %d\n", ret);
```

**Review by martinjaeger — CHANGES_REQUESTED** [review]

**Review by martinjaeger — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Pull request merged**